File: CS50x/pset6/readability/readability.py
from cs50 import get_string


def Main():

    # get user Input
    text = get_string("Text: ")

    # variables
    s = 0.0
    l = 0.0

    l = CheckLetter(text)
    s = CheckSentences(text)
    
    index = ((0.0588 * l) - (0.296 * s) - 15.8)
    
    if index < 1:
        index = 1
        
    if index > 16:
        print("Grade 16+")
    elif index < 2:
        print("Before Grade 1")
    else:
        print("Grade", int(round(index)))

    
def CheckLetter(text):
    # letter and words
    letter = 0
    words = 1
    # adding words and letters
    for i in range(0, len(text), 1):
        if text[i] == ' ':
            words += 1
        elif not CheckTable(text[i], False):
            letter += 1
            # print(text[i], letter)
    # return letter avarage
    # print((float(letter) / words * (100-1.5)))
    return (float(letter) / words * 100)
    

def CheckSentences(text):
    # getting sentence
    # getting words
    # period_check is to make sure double .(U.S.)
    sentences = 0
    word = 1
    periodCheck = -1
    secondPeCheck = 0

    for i in range(0, len(text), 1):
        if text[i] == " ":
            word += 1
        elif CheckTable(text[i], True):
            # period checking
            if i + 2 < len(text):
                secondPeCheck = i + 2
            else:
                secondPeCheck = 0
                
            if text[i] == "." and text[secondPeCheck] == "." and secondPeCheck != i:
                periodCheck = i + 2
            elif periodCheck < i:
                sentences += 1
    # print((float(sentences) / word * 100))
    return (float(sentences) / word * 100)
    

def CheckTable(char, isSentence):
    # checking the table
    # i hate the face that
    # switch case don't exist...
    listA = ['!', '?', '.']
    listB = [',', ':', ';', '\'', '\0']
    if isSentence == True:
        if char in listA:
            return True
    else:
        if char in listB or char in listA:
            return True
    # else return false
    return False

    
Main()

File: CS50x/pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

float check_sentences(int length, string str);
float check_letter(int length, string str);
bool check_table(char ch, bool is_sentence);

int main(void)
{
    string text = get_string("Text: ");
    
    //getting length of string
    //setting up l and s (L, S)
    int length = 0;
    float l = 0.0f, s = 0.0f;
    
    //length
    for (length = 0; text[length] != '\0'; ++length);
    
    l = check_letter(length, text);
    s = check_sentences(length, text);
    
    float index = ((0.0588 * l) - (0.296 * s) - 15.8);
    
    //making sure that the grade is atlest 1
    index = index < 1 ? 1 : index;
    
    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 2)
    {
        printf("Before Grade 1\n");
    }
    else 
    {
        printf("Grade %i\n", (int) round(index));
    }
}

float check_sentences(int length, string str)
{
    /*getting sentence
     *getting words
     *period_check is to make sure double .(U.S.)
     */
    int sentence = 0;
    int words = 1;
    int period_check = -1;
    
    for (int i = 0; i < length; ++ i)
    {
        if (str[i] == ' ')
        {
            ++words;
        }
        else if (check_table(str[i], true))
        {
            /*period_checking here*/
            if (str[i] == '.' && str[i + 2] == '.')
            {
                period_check  = i + 2;
            }
            
            if (period_check < i)
            {
                sentence ++;
            }

        }
    }
    return ((float)sentence / words * 100);
}

float check_letter(int length, string str)
{
    //letters and words
    int letters = 0;
    int words = 1;
    
    //adding to letters and word
    for (int i = 0; i < length; ++ i)
    {
        if (str[i] == ' ')
        {
            ++words;
        }
        else if (!check_table(str[i], false))
        {
            ++letters;
        }
    }

    return ((float) letters / words * 100);
}

bool check_table(char ch, bool is_sentence)
{
    //switch case that for some reason 
    //fixed many of my bugs after 
    //looking at the example to check through a sentence counter
    switch (ch)
    {
        case '!':
        case '?':
        case '.':
            if (is_sentence)
            {
                return true;
            }
        case ',':
        case ':':
        case ';':
        case '\'':
        case '\0':
            if (!is_sentence)
            {
                return true;
            }
    }
    
    return false;
}